core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -436,21 +436,24 @@ private CompletableSource sendUpdateBatch(Collection<? extends IracStateData> ba
             cmd.addUpdate(data.state.getKey(), data.entry.getValue(), data.entry.getMetadata(), data.entry.getInternalMetadata().iracMetadata());
          }
       }
-      IracResponseCollector rspCollector = new IracResponseCollector(commandsFactory.getCacheName(), validState, this::onBatchResponse);
-      try {
-         for (IracXSiteBackup backup : asyncBackups) {
-            if (takeOfflineManager.getSiteState(backup.getSiteName()) == SiteState.OFFLINE) {
-               continue; // backup is offline
+
+      IracResponseCollector rspCollector = null;
+      if (!cmd.isEmpty()) {
+         rspCollector = new IracResponseCollector(commandsFactory.getCacheName(), validState, this::onBatchResponse);
+         try {
+            for (IracXSiteBackup backup : asyncBackups) {
+               if (takeOfflineManager.getSiteState(backup.getSiteName()) == SiteState.OFFLINE) {
+                  continue; // backup is offline
+               }
+               rspCollector.dependsOn(backup, sendToRemoteSite(backup, cmd));
             }
-            rspCollector.dependsOn(backup, sendToRemoteSite(backup, cmd));
-         }
-      } catch (Throwable throwable) {
-         // marshalling error? JGroups error?
-         for (IracStateData data : batch) {
-            data.state.retry();
+         } catch (Throwable throwable) {
+            // marshalling error? JGroups error?
+            for (IracStateData data : batch) {
+               data.state.retry();
+            }
+            onUnexpectedThrowable(throwable);
          }
-         onUnexpectedThrowable(throwable);
-         return Completable.complete();
```

Comment:
This slightly changes how exception handling operates. With the new changes if there is a Throwable, we now have to wait for the registered remote site backup commands to complete. Do we want that? It is probably a very small chance of this happening, but just wanted to make sure.

Reply:
Not quite, If you have more than one backup if the second errors, we would have to wait until the first completes as rspCollector would have a stage dependency.

Reply:
I see. I'm setting `rspCollector = null;` to avoid waiting although, I think, this code will never be executed.

Reply:
Oh, yes I missed that sorry. So we are good.